pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -136,6 +141,34 @@ func newAlertmanager(orgID int64, cfg *setting.Cfg, store store.AlertingStore, k
 		return nil, err
 	}
 
+	// Setup clustering.
+	// TODO: Move me
+	if len(cfg.HAPeers) > 0 {
+		peer, err := cluster.Create(
+			gokit_log.With(am.gokitLogger, "component", "cluster"),
+			m.Registerer,
+			cfg.HAListenAddr,
+			cfg.HAAdvertiseAddr,
+			cfg.HAPeers, // peers
+			true,
+			cfg.HAPushPullInterval,
+			cfg.HAGossipInterval,
+			cluster.DefaultTcpTimeout,
+			cluster.DefaultProbeTimeout,
+			cluster.DefaultProbeInterval,
+		)
+
+		if err != nil {
+			return nil, fmt.Errorf("unable to initialize gossip mesh: %s", err)
+		}
+
+		err = peer.Join(cluster.DefaultReconnectInterval, cluster.DefaultReconnectTimeout)
+		if err != nil {
+			am.logger.Error("msg", "unable to join gossip mesh while initializing cluster for high availability mode", "err", err)
+		}
+		go peer.Settle(context.Background(), cluster.DefaultGossipInterval)
```

Comment:
Do we want to be able to cancel this context?

Reply:
Good spot! You want to take a look at what I've done + the explanation?